ArduPlane/ArduPlane.cpp

```diff
@@ -968,6 +969,81 @@ void Plane::update_flight_stage(void)
     airspeed.set_EAS2TAS(barometer.get_EAS2TAS());
 }
 
+/*
+*  ArduSoar support function
+*
+*  Peter Braswell and Samuel Tabor
+*/
+void Plane::update_soaring() {
+    
+    if (!soaring_controller.is_active()) {
+        return;
```

Comment:
It did take some work to ensure correct behaviour under the various thermalling start/end scenarios. I have covered them pretty thoroughly in my testing, with the possible exception of changing the SOAR_ACTIVE parameter. Hopefully I can repeat the tests in simulation using this branch. The variables you mentioned can be initialised to zero.